### DISCUSSION TITLE
Docking denied - Limit message spamming / repetition

### DISCUSSION THREAD

**Discussion by Thormor** [EDCD/EDDI#2131] [Troubleshoot Scripts] [answered]
Hello!
I wanted to limit the number of message queued (with speech responder) especially in the case of docking denied. When trying to land a busy platforms (like a rescue ship for i.e.) hearing for 5 minutes the docking denied [reason] is a tad annoying. I wanted to modify the script to have a limit of how many of those it will play in a given minute. Any suggestions on how to do it cleanly?

**Comment by Tkael** [comment, marked as answer]
I see three options:

1. If you can't find a place to dock at a busy port or station then consider switching game modes. The wait will be much shorter if you switch to solo (even if only as a temporary measure to find a pad). :-)

2. There is actually a state context variable which is automatically saved containing the last speech EDDI has spoken. It is `state.eddi_context_last_speech`. You could potentially check this value for a repetition. This is a simple solution which would be reset if EDDI triggers any other speech during that 5 minute period, but it could be done by adding one line at the beginning of the script:
```
{if find(state.eddi_context_last_speech, "We have been denied docking") > -1: {return ""} }
```

3. You can set some more specific context variables and respond to those. For example, you could add this to the beginning of the script:
```
{if state.eddi_context_last_subject = "docking" && state.eddi_context_last_action = "denied": {return ""} }
```

And this to the end of the script:
```
{SetState("eddi_context_last_subject", "docking")}
{SetState("eddi_context_last_action", "denied")}
```

The final script might look like this:
```
{if state.eddi_context_last_subject = "docking" && state.eddi_context_last_action = "denied" : {return ''} }

{Pause(2000)}

We have been denied docking

{if event.reason = "ActiveFighter":
    because our fighter is deployed
|elif event.reason = "Distance":
    because we are too far from the station
|elif event.reason = "Hostile":
    because the station's controlling faction doesn't like us
|elif event.reason = "NoSpace":
    because there is no space available
|elif event.reason = "Offences":
    because we have recently committed an offence against them
|elif event.reason = "TooLarge":
    because there are no landing pads that can accomodate the size of our ship
}.

{SetState('eddi_context_last_subject', 'docking')}
{SetState('eddi_context_last_action', 'denied')}
```

Naturally, these strategies can be used for any other scripts where you might want to suppress repetitions too. o7
